### PR TITLE
HPCC-23858 Install aws/yaml libraries if not PLUGIN build

### PR DESCRIPTION
**Pull request by Michael-Gardner**
Signed-off-by: Michael Gardner <michael.gardner@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@xwang2713 Please review.  Can you test this against the windows builds to ensure everything builds and works correctly?  This ensures that aws and yaml get installed as part of the platform (other platform libraries like jlib are linking against them) and we don't attempt to re'install them during a plugins build.  Tested on 18.04 and things seem to be linking correctly.

File: system/aws/CMakeLists.txt [hunk 1]
```diff
@@ -33,30 +33,31 @@ if(USE_AWS)
...
-        DESTINATION lib)
-
+    if ( NOT PLUGIN )
+        install(CODE "set(ENV{LD_LIBRARY_PATH} \"\$ENV{LD_LIBRARY_PATH}:${AWS_DEP_LIBDIR}:${PROJECT_BINARY_DIR}:${PROJECT_BINARY_DIR}/aws-sdk-cpp/aws-cpp-sdk-core:${PROJECT_BINARY_DIR}/aws-sdk-cpp/aws-cpp-sdk-sqs:${PROJECT_BINARY_DIR}/aws-sdk-cpp/aws-cpp-sdk-s3\")")
+        install(FILES
+            ${CMAKE_CURRENT_SOURCE_DIR}/aws-sdk-cpp/LICENSE.txt
+            DESTINATION lib
+            RENAME aws-sdk-cpp-LICENSE.txt)
+        install(CODE "file(RPATH_CHANGE
+            FILE \"${CMAKE_CURRENT_BINARY_DIR}/aws-sdk-cpp/aws-cpp-sdk-core/libaws-cpp-sdk-core.so\"
```
Comment (richardkchapman) [inline system/aws/CMakeLists.txt:43]: .so extensions are not correct for OSX (or windows - which also probably does not want the lib prefix)

Comment (Michael-Gardner) [reply system/aws/CMakeLists.txt:43]: I'll change them to use the same lib suffix variable I used in the yaml library code.

File: system/yaml/CMakeLists.txt [hunk 2]
```diff
@@ -16,11 +16,29 @@
 
 project( yaml_include)
 
-remove_definitions(-fvisibility=hidden)
-
 if(WIN32)
-    set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+    set(LIBYAML_OUTPUT_DIR ${CMAKE_BINARY_DIR}/bin)
+    add_subdirectory(${HPCC_SOURCE_DIR}/system/yaml/libyaml ${CMAKE_BINARY_DIR}/system/libyaml)
```
Comment (Michael-Gardner) [inline system/yaml/CMakeLists.txt:21]: @xwang2713 we'll keep it the old way for windows clienttools builds.  Only need the new logic for linux/osx.

**Comment by Michael-Gardner** [comment]
@xwang2713 fixed the windows build issue we were running into.

File: system/yaml/CMakeLists.txt (hunk 2 above)
Comment (xwang2713) [reply system/yaml/CMakeLists.txt:23]: OK

**Comment by xwang2713** [comment]
Looks good

**Comment by Michael-Gardner** [comment]
@richardkchapman ready for merge

**Pull request merged**